### ISSUE TITLE
Match on WM_CLASS matches substrings

### ISSUE THREAD

**Issue by Sawrz** [qtile/qtile#2670]
# Issue description

It seems the `Match` class checks if a sub-string exists in in a given `wm-class` string. For example:

Sublime Text's `WM_CLASS` is `subl`. Now, I want to have a intrusive float match for `Ssh-askpass-sublime`. However adding that match makes Sublime Text an intrusive floating windows as well. 

I tried different combinations, like `sublome` but the result is the same. If I understood the source code correctly you use a lambda function to compare two strings with the `in` keyword. However, I didn't dig to much into the code so I  might be wrong. 

Is that on purpose? 

# Qtile version

`0.18.1.dev0+g8e7ecc0a.d20210719 `
Seems to be Version 0.18.0 according to releases tab

# Configuration

[My configuration file](https://gitlab.com/lazylinuxuser/dotfiles/-/blob/main/qtile/.config/qtile/config.py)

**Comment by yobleck** [comment]
Yes the Match class does appear to be using substring matching [here](https://github.com/qtile/qtile/blob/a3dcd5db984f3ab08ef3f89eff86e014dd367ee1/libqtile/config.py#L605)

This is just a temporary hack but you could try:
```
@hook.subscribe.client_managed
    def my_func(window)
        window.window.get_wm_class() #or get name or something
        #some logic to parse the correct window the way you want
        qtile.cmd_spawn("xdotool search --name "Title of App" set_window --class "New WM Class" --classname "New WM Class Name")
        #requires xdotool from your package manager
```

([xdotool man page](https://manpages.ubuntu.com/manpages/trusty/man1/xdotool.1.html))

**Comment by Sawrz** [comment]
Thank you @yobleck!

**Comment by m-col** [comment]
The bit from the docs about that "func" argument: "delegate the match to the given function, which receives the tested client as argument and must return True if it matches, False otherwise". In the function you provide you can specify the logic yourself, so for example something like this might testing your askpass window and only that window (untested):
```py
mymatch = Match(func=lambda w: w.get_wm_class() == "Ssh-askpass-sublime")
```

**Comment by Sawrz** [comment]
@m-col I saw that argument as well. I didn't used it for two reasons:
1. I wasn't sure how to use it
2. I don't know if this behavior, as implemented, is intended

Anyway, no harm in trying it out. However, it doesn't seem to work. To check it's validity I tried to force `subl` in my match to see what happens. The result is: Nothing. No intrusive or floating behavior, so we seem to miss something here.

I tried to get the `w.get_wm_class()` output in a `txt`-file but qtile crashed and no  output in my file.
